### PR TITLE
Fix stripe API rate limit error

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
## Description

This pull request fixes a regression issue where hitting the stripe API rate limit would result in an error. The issue has been resolved by removing the unnecessary `stripePortalUrl` state and updating the code accordingly.

regression but introduced in #19417

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/AppNotifications.tsx [hunk 1]
```diff
@@ -113,10 +112,10 @@ export function AppNotifications() {
                     const [subscriptionId, invalidBillingAddress] = await Promise.all([
                         getGitpodService().server.findStripeSubscriptionId(attributionId),
                         getGitpodService().server.isCustomerBillingAddressInvalid(attributionId),
-                        getGitpodService().server.getStripePortalUrl(attributionId).then(setStripePortalUrl),
```
Comment (jeanp413) [inline components/dashboard/src/AppNotifications.tsx:116]: Keep the `getStripePortalUrl`  call here so it's awaited at the same time just remove the `then`

**Review by jeanp413 — APPROVED** [review]

File: components/dashboard/src/AppNotifications.tsx (hunk 1 above)
Comment (Siddhant-K-code) [reply components/dashboard/src/AppNotifications.tsx:116]: but, then will it update the value to `stripePortalUrl`?

Comment (Siddhant-K-code) [reply components/dashboard/src/AppNotifications.tsx:116]: resolved in [1cc320d](https://github.com/gitpod-io/gitpod/pull/19431/commits/1cc320df184a81ed51efa012a802ec15ec92de42)

**Pull request merged**